Documentation/ApiOverview/PasswordHashing/Index.rst

```diff
@@ -13,54 +13,54 @@ Password Hashing
+the attacker has most likely gained access to more than just the user's TYPO3 Frontend / Backend account.
+It's not uncommon for someone to use the same email and password combination for more than one online service,
+such as their bank account, social media accounts, or even their email provider.
+
+To mitigate this risk, we can use one-way `hash`_ algorithms to transform a plain text password into an incomprehensible
+string of seemingly "random" characters.
+A hash, (also known as a checksum), is the result of a value (in this case the user's password) that has been
+mathematically calculated using a one-way function.
+Hashes are called "one-way functions" because they're quick and easy to generate, but incredibly hard to undo.
+You would require an incredible amount of computational power and time to try and reverse a hash back to its original
+value.
+
+When a user tries to log in and submits their password through the login form, the same one-way function is performed
+and the result is compared against the hash stored in the database. If they're the same, we can safely assume the user
+typed in the correct password without ever needing to store their actual password!
+
+The most well-known hash algorithm is MD5. Basic hash algorithms and especially MD5 have drawbacks though:
+First, if you find some other string that resolves to the same hash, you're screwed (that's called
 a collision). An attacker could login with a password that is not identical to "your" password, but still matches
 the calculated hash. And second, if an attacker just calculates a huge list of all possible passwords with their
 matching hashes (this is called a rainbow table) and puts them into a database to compare any given hash with,
 it can easily look up plain text passwords for given hashes. A simple MD5 hash is susceptible to both of these
 attack vectors and thus deemed insecure. MD5 rainbow tables for casual passwords can be found online and MD5 collision
-creation can be done without too many issues. In short: MD5 is not a good idea to secure user passwords.
+creation can be done without too many issues. In short: MD5 is not a suitable hashing algorithm for securing user passwords.
 
 To mitigate the rainbow table attack vector, the idea of "salting" has been invented: Instead of
 hashing the given password directly and always ending up with the same hash for the same password
-(if different users use the same password they end up with the same hash in database), a "salt"
+(if different users use the same password they end up with the same hash in the database), a "salt"
 is added to the password. This salt is a random string calculated when the password is first set (when the
 user record is created) and stored *together* with the hash. The basic thinking is that the salt is
 appended to the password before hashing, the "salt+password" combination is then hashed. The salt is stored
```

Comment:
We should either write "prepended" (that's the common way) or simply "added" just like above.

Reply:
I think it might be better to use "added" since it doesn't assume a particular pattern, even if prepending a salt is the most common pattern used.